### DISCUSSION TITLE
HDR相关的初次配置 求大佬指导

### DISCUSSION THREAD

**Discussion by Hyphen-H** [dyphire/mpv-config#26] [Q&A] [answered]
最近购入了R9000P, 显卡是4060，支持HDR, 手有点痒又想折腾了，请问应该如何以正确姿势播放诸如HDR.H265这样的文件？
mpv使用的portable_config是dyphire/mpv-config, MPV.exe来自dyphire/mpv-config的release下的mpv_config-latest包
请问HDR映射到SDR在这套config下是自动开启的吗？我需要关闭映射曲线吗？新手小白不知道如何设置才是最正确的姿势。。

系统是win11，windows的HDR需要开启吗？初次接触HDR好迷茫。。

求大佬指导，感谢！

**Comment by dyphire** [comment, marked as answer]
github都不推送通知的，才看到

> 请问HDR映射到SDR在这套config下是自动开启的吗？

是的，无论是此配置还是mpv默认值都是会自动启用HDR映射的

> 我需要关闭映射曲线吗？

如果启用HDR直通的话将自动忽略映射曲线

> 系统是win11，windows的HDR需要开启吗？

视使用的gpu-api后端而定

关于HDR直通的相关说明在我的相应配置组中有一些对应的内容

https://github.com/dyphire/mpv-config/blob/58dbda429f5bcfac90f0b2694a02a8b5ed50bd2f/mpv.conf#L543-L545

顺便HDR显示器最好在HDR600及以上才能有相对正常的HDR画面表现，否则不建议开启HDR直通

**Reply by Hyphen-H** [reply]
屏幕说是有杜比视界认证🤣 效果怎样我先通一通看看><
那我的配置就这样写吗：
```
profile=HDR-PASS                             # HDR直通参数配置组，目前只能在初始化时设置，HDR直通时需禁用--icc-profile,icc-profile-auto和--target-trc参数。--vo=gpu-next --gpu-api=vulkan下和全屏（Windows的限制）时可用，--gpu-api=d3d11下需手动切换显示器HDR模式（Windows的限制） 
 #profile=SDR2HDR                              # SDR映射HDR参数配置组，仅在hdr设备上有效，不推荐启用。仅限--vo=gpu-next下可用 
 profile=HDR2SDR 
```
然后
![image](https://github.com/dyphire/mpv-config/assets/103420806/6e4f962e-53dc-4c75-bd79-b1fdf60374ae)

**Reply by dyphire** [reply]
请参考主仓库最新配置，你图里的配置很老了，已经过时了。主仓库配置有很多改动和修复

杜比视界内容无法直通，只能重塑为SDR内容
正常的HDR内容直通需启用HDR-PASS配置组并将gpu-api切换为vulkan
由于HDR直通参数`--target-colorspace-hint`只能在初始化时配置，无法通过条件配置应用（mpv的限制）。而此项无法和`--icc-profile*`还有`--target-*`修改项共存。~这就意味着HDR内容直通和正确的SDR内容显示不可兼得（后者需要`--icc-profile*`或`--target-trc`等参数来配置正确的屏幕特性）~

忘记之前已经想到过共存的处理方法了：https://github.com/dyphire/mpv-config/commit/764d104d3906234f6811d87672e3d6dc2cbbd00c
启用HDR直通配置组HDR-PASS时需将HDR条件配置组中的相关注释取消掉，效果为：
```ini
[HDR]
 profile-desc=HDR配置
 profile-cond=p["video-params/sig-peak"] > 1
 profile-restore=copy
 icc-profile=""
 icc-profile-auto=no
 target-trc=auto
 #target-peak=100
 blend-subtitles=no
 deband=no
 vf=""
```
